examples/apps/ai-collab/src/app/presence.ts

```diff
@@ -0,0 +1,126 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	IPresence,
+	Latest,
+	type ISessionClient,
+	type PresenceStates,
+	type PresenceStatesEntries,
+	type PresenceStatesSchema,
+} from "@fluidframework/presence/alpha";
+
+import { getProfilePhoto } from "@/infra/authHelper";
+
+export interface User {
+	photo: string;
+}
+
+const statesSchema = {
+	onlineUsers: Latest({ photo: "" } satisfies User),
+} satisfies PresenceStatesSchema;
+
+export type UserPresence = PresenceStates<typeof statesSchema>;
+
+// Takes a presence object and returns the user presence object that contains the shared object states
+export function buildUserPresence(presence: IPresence): UserPresence {
+	const states = presence.getStates(`name:user-avatar-states`, statesSchema);
+	return states;
+}
+
+export class PresenceManager {
+	// A PresenceState object to manage the presence of users within the app
+	private readonly usersState: UserPresence;
+	// A map of SessionClient to UserInfo, where users can share their info with other users
+	private readonly userInfoMap: Map<ISessionClient, User> = new Map();
+	// A callback methid to get updates when remote UserInfo changes
+	private userInfoCallback: (userInfoMap: Map<ISessionClient, User>) => void = () => {};
+
+	constructor(private readonly presence: IPresence) {
+		this.presence = presence;
```

Comment:
Not needed. Since the constructor parameter is already declared as `private readonly presence` and not just `presence`, that already assigns it to the private property
```suggestion
```